src/librustc_mir/build/expr/category.rs

```diff
@@ -43,7 +43,9 @@ pub enum RvalueFunc {
 impl Category {
     pub fn of<'tcx>(ek: &ExprKind<'tcx>) -> Option<Category> {
         match *ek {
-            ExprKind::Scope { .. } => None,
+            ExprKind::Scope { .. }
+            | ExprKind::PlaceTypeAscription { .. }
+            | ExprKind::ValueTypeAscription { .. } => None,
```

Comment:
These should not return `None` -- rather I think they should both return `Some(Category::Place)`.